### PR TITLE
Add mStable to dex.trades in Dune v2 for Ethereum Mainnet.

### PR DESCRIPTION
**Pull request by ripples3**
Brief comments on the purpose of your changes:

*For Dune Engine V2*
I've checked that:
General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributors)

Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
* [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by ripples3** [comment]
related to https://github.com/duneanalytics/spellbook/issues/1660

**Comment by jeff-dude** [comment]
> There is a blocker for this merge. Please see https://discord.com/channels/757637422384283659/850326962152538122/1027556527483138189

is this still blocked?

**Comment by ripples3** [comment]
Still blocked. I will follow-up the decoding team again this week.

**Comment by ripples3** [comment]
@jeff-dude, should be good now.

The queries below are for checking
Test Query
https://dune.com/queries/1500288?d=1

Comparison between v1 and v2
https://dune.com/queries/1358505
https://dune.com/queries/1358520

File: models/mstable/ethereum/mstable_ethereum_trades.sql [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            '1' AS version,
+            'DEX' AS category,
+            swapper AS trader_a,
+            cast(NULL as string) AS trader_b,
+            `outputAmount` AS token_a_amount_raw,
+            cast(NULL as double) AS token_b_amount_raw,
+            CASE WHEN `output` = lower('0x0000000000000000000000000000000000000000') THEN 
+                lower('0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE') ELSE `output`
+                END AS token_a_address,
+            CASE WHEN `input` = lower('0x0000000000000000000000000000000000000000') THEN
```
Comment (0xBoxer) [inline models/mstable/ethereum/mstable_ethereum_trades.sql:97]: This does not work in V2, need to replace with WETH address

File: models/mstable/ethereum/mstable_ethereum_trades.sql [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            'DEX' AS category,
+            swapper AS trader_a, 
+            cast(NULL as string) AS trader_b,
+            `outputAmount` AS token_a_amount_raw,
+            cast(NULL as double) AS token_b_amount_raw,
+            CASE WHEN `output` = lower('0x0000000000000000000000000000000000000000') THEN 
+                lower('0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE') ELSE `output`
+                END AS token_a_address,
+            CASE WHEN `input` = lower('0x0000000000000000000000000000000000000000') THEN 
+                lower('0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE') ELSE `input`
```
Comment (0xBoxer) [inline models/mstable/ethereum/mstable_ethereum_trades.sql:69]: this does not work in V2, need to work with WETH address

**Review by 0xBoxer — CHANGES_REQUESTED** [review]

File: models/mstable/ethereum/mstable_ethereum_schema.yml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      - &token_bought_amount_raw
+        name: token_bought_amount_raw
+        description: "Raw value of the token bought at time of execution in the original currency"
+      - &token_sold_amount_raw
+        name: token_sold_amount_raw
+        description: "Raw value of the token sold at time of execution in the original currency"
+      - &amount_usd
+        name: amount_usd
+        description: "USD value of the trade at time of execution"
+      - &token_bought_address
```
Comment (jeff-dude) [inline models/mstable/ethereum/mstable_ethereum_schema.yml:64]: please add the dex generic test here on this column to validate bought address:
https://github.com/duneanalytics/spellbook/pull/1866/files#diff-8ba1908b2679efbd3dc591b730739b0c156cd7fe53a264c85c172ef2762d3205R64

for this test to run, you'll need to manually add rows to our seed file:
https://github.com/duneanalytics/spellbook/pull/1866/files#diff-09c1b2e11432d6907b7f05a0e73f46488de55e9f905df8d54602cb7173fdba5fR55

these are examples from a recently merged PR. some protocols offer an analytics page on the app, which makes it easy to find tx's which link out to etherscan. the goal is to find tx's on a third-party source and validate against the query results.

**Review by jeff-dude — CHANGES_REQUESTED** [review]

**Comment by jeff-dude** [comment]
please do look at the recent commits as well to see patterns around formatting/naming/etc. and such to maintain consistency across dex models

File: models/mstable/ethereum/mstable_ethereum_schema.yml (hunk 3 above)
Comment (ripples3) [reply models/mstable/ethereum/mstable_ethereum_schema.yml:64]: completed this request.

**Comment by ripples3** [comment]
Hi @jeff-dude, I committed the new changes.

**Comment by jeff-dude** [comment]
@ripples3 if you compile the code for the tests added, you'll get this query (note this is my local test tables, you could also replace with tables built via gh action):
```
with
  unit_test as (
    select
      case
        when lower(m.token_bought_address) = lower(seed.token_bought_address) then True
        else False
      end as token_bought_test
    from
      dbt_jeff_mstable_ethereum.trades m
      inner join dbt_jeff_test_data.dex_trades_seed seed on m.tx_hash = seed.tx_hash
      and m.evt_index = seed.evt_index
      and m.block_date = seed.block_date
      and m.blockchain = seed.blockchain
      and m.project = seed.project
      and m.version = seed.version
  )
select
  *
from
  unit_test
-- where
--   token_bought_test = False
```

i commented out the filter for false to ensure all four rows in the seed file are returned in the join above. it looks like none return, which means there is likely bad data in the seed file. please edit the seed data until this query returns all four rows (likely using the gh action tables instead, as they'll update with your commits/gh action running)

**Comment by ripples3** [comment]
The issue was caused by the project name.  In the file `mstable_ethereum_trades.sql` it was `mStable` while in the seed file it is `mstable`. 
I already fixed it.
Test in Dune returns 4 rows now.  https://dune.com/queries/1591728?d=1

Here is the screenshot 
image

**Review by jeff-dude — APPROVED** [review]
awesome, great stuff @ripples3 🤝

**Pull request merged**